modules/core/src/main/kotlin/datamaintain/core/step/executor/Executor.kt

```diff
@@ -121,8 +121,35 @@ class Executor(private val context: Context) {
         try {
             context.dbDriver.overrideScript(it)
         } catch (e: Exception) {
-            if (!context.config.porcelain) { logger.error { "error during override of ${it.fullName()} " } }
+            if (!context.config.porcelain) {
+                logger.error { "error during override of ${it.fullName()} " }
+            }
             throw e
         }
     }
 }
+
+fun buildSimulatedExecutedScript(script: ScriptWithContent, executionStatus: ExecutionStatus, flags: List<String>) =
+    ExecutedScript(
+        script.name,
+        script.checksum,
+        script.identifier,
+        executionStatus,
+        script.action,
+        flags = flags
+    )
+
+fun buildExecutedScript(script: ScriptWithContent, execution: Execution, flags: List<String>) =
+    buildSimulatedExecutedScript(script, execution.executionStatus, flags)
+
+fun buildExecutedScript(script: ScriptWithContent, execution: Execution, executionDurationInMillis: Long, flags: List<String>) =
+    ExecutedScript(
+        script.name,
+        script.checksum,
+        script.identifier,
+        execution.executionStatus,
+        script.action,
+        executionDurationInMillis,
+        execution.executionOutput,
+        flags = flags
+    )
```

Comment:
An extends function seem better

Reply:
Can't extend a companion object that does not exist, I would need to create an empty one in the domain.